### ISSUE TITLE
Tags are not removed when project removed

### ISSUE THREAD

**Issue by kiselli** [kanboard/kanboard#3231]
### Actual behaviour

When I remove a project and create a new one (it gets the same ID as the removed one) the project has the same tags as the previously removed project.

### Expected behaviour

When a project is removed, all associated tags should be removed or at least the association.

### Steps to reproduce

1. Log in to the application 
2. Create a new Project
3. Create a Tag
4. Remove Project
5. Create a new Project
6. Take a look at Tags

### Configuration

Copy and paste the configuration section from the Kanboard settings page or supply values for these fields:
```
Application version: 1.0.41
PHP version: 7.0.17-2+deb.sury.org~xenial+1
PHP SAPI: fpm-fcgi
OS version: Linux 4.4.0-66-generic
Database driver: mysql
Database version: 10.0.29-MariaDB-0ubuntu0.16.04.1
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36
```

**Comment by rautamiekka** [comment]
Can you
1) duplicate your database
2) update that duplicate to the current Kanboard by downloading the repo
3) try reproducing
?

**Comment by kiselli** [comment]
I did another test with the current dev version in a docker environment. 

```
Application version: master
PHP version: 7.0.15-0ubuntu0.16.04.4
PHP SAPI: fpm-fcgi
OS version: Linux 4.9.13-moby
Database driver: sqlite
Database version: 3.11.0
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36
```

The Issue keeps the same. 

**I did a few tests and fixed the problem to the sqlite autoincrement strategy.**

You can read about that in the sqlite AUTOINCREMENT docs:


> The usual algorithm is to give the newly created row a ROWID that is one larger than the largest  ROWID in the table prior to the insert. If the table is initially empty, then a ROWID of 1 is used. If the largest ROWID is equal to the largest possible integer (9223372036854775807) then the database engine starts picking positive candidate ROWIDs at random until it finds one that is not previously used. If no unused ROWID can be found after a reasonable number of attempts, the insert operation fails with an SQLITE_FULL error.


Therefore the new Project gets the same ID as the previously deleted one, if there was no new project created in beteen.

I also had a look at the source code, seems there is no code to "clean" delete referenced items when a project is removed.

`Kanboard\Controller\ProjectStatusController`
```
/**
     * Remove a project
     */
    public function remove()
    {
        $project = $this->getProject();
        $this->checkCSRFParam();

        if ($this->projectModel->remove($project['id'])) {
            $this->flash->success(t('Project removed successfully.'));
        } else {
            $this->flash->failure(t('Unable to remove this project.'));
        }

        $this->response->redirect($this->helper->url->to('ProjectListController', 'show'));
    }
```

`Kanboard\Model\ProjectModel`

```
/**
     * Remove a project
     *
     * @access public
     * @param  integer   $project_id    Project id
     * @return bool
     */
    public function remove($project_id)
    {
        return $this->db->table(self::TABLE)->eq('id', $project_id)->remove();
    }
```

**Comment by kiselli** [comment]
I created some Tests to validate the suggested behaviour.

https://github.com/kanboard/kanboard/pull/3250

**Comment by rautamiekka** [comment]
Do we have a **real reason** to worry about an extreme number like 9223372036854775807 ? If you're using the autoincrement column the way you're meant to, there's just absolutely no way to reach that number, especially in an SQLite db. Also, I'm not aware of any means to jump the autoincrement in steps other than +1 (think you could use your own number during `INSERT` or `UPDATE`, but that's just not sensible). Even if you jumped the increment, you'd likewise have to use extreme numbers to reach it, and I just don't see any sorta sense in that.

**Comment by fguillot** [comment]
@kiselli Project tags are now removed when a project is removed (in master).

**Comment by fguillot** [comment]
Fix available in v1.0.44.
